### ISSUE TITLE
Async resources providers in list providers

### ISSUE THREAD

**Issue by mxab** [ets-labs/python-dependency-injector#450]
Hi,
I'm trying to use async resources with list providers but it doesn't seem to work

```python
from typing import List
from dependency_injector import containers,providers
import asyncio
class MyResource:
    def __init__(self,my_param:str) -> None:
        self.my_param = my_param

async def create_resource(param:str):
    return MyResource(my_param=param)


class Printer:
    def __init__(self, resources: List[MyResource]) -> None:
        self.resources = resources
    def print(self):
        for r in self.resources:
            print(r.my_param)

class Container(containers.DeclarativeContainer):

    my_resources = providers.List(
        providers.Resource(create_resource, "foo"),
        providers.Resource(create_resource, "bar")
        )
    
    printer = providers.Factory(Printer, resources = my_resources)


async def main():
    container = Container()
    printer = await container.printer()

    printer.print()


if __name__ == "__main__":
    asyncio.get_event_loop().run_until_complete(main())
```

if I remove the `async` from the `create_resource` function and the `await` infront of `container.printer()` it works.

Is there anything I missed or is it currently not supported?

**Comment by rmk135** [comment]
Hey @mxab ,

Yeah, this is a bug. Thanks for reporting this. Working on the fix.

**Comment by rmk135** [comment]
@mxab Fixed in `4.32.1`.  Thanks again for bringing this up.

**Comment by mxab** [comment]
Nice! Will try today
